### PR TITLE
Products by Rating: Add display options

### LINKED ISSUE #7171: Products by Rating: Add support for Display Options
Edit_Page_“All_Products”_‹_ratings_—_WordPress

The Products by Rating filter block should follow the Display Options pattern of other filter blocks.

Related issues: https://github.com/woocommerce/woocommerce/issues/42588

### PR DESCRIPTION
**Pull request by danieldudzic**
Add support for the following display options:
- Product count
- 'Apply filters' button
- Typography - Size
- Color 

Fixes #7171 

### Screenshots
190835898-f62d5dff-8921-44ea-acf2-d5f690544ef7

### Testing
1. Activate a block theme & setup WooCommerce with products that have some ratings
2. Create a classic shop page and a separate page using the All Products block
3. Add the `Filter Products by Rating` block and `Active Filters` block on both pages
4. Verify that all Display Options display and work correctly both in the editor and the page

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

### Changelog

> Products by Rating: Add support for Display Options

### CODE REVIEW

**Comment by tjcafferkey** [comment]
Just a few things I've noticed.

1. When I add the filter to a template Site Editor > All Templates > Product Catalog I don't get the icons in the Editor like I do on a regular page alongside the All Products block. The icons work on the frontend though.

Screenshot 2022-10-07 at 15 15 13

2. I can see under the Color settings panel there is an option to select a color for "Link" but I'm not sure which part of the block this applies to as when I select a color nothing changes visually?

Screenshot 2022-10-07 at 15 26 35

3. When I use this filter alongside the Filter By Attribute (color) block and then select an attribute to filter by, select "Apply". Then the "Filter by Rating" block is the only block which goes back into a loading state. When I reverse this action and select a rating to filter by, the Filter By Attribute block _does not_ go into a loading state. I think **we need to stop the placeholders from rerendering** for the rating block every time we select an attribute from the Filter by Attribute block.

File: assets/js/blocks/rating-filter/style.scss [hunk 1]
```diff
@@ -24,3 +24,29 @@
...
+
+.wc-block-rating-filter__actions {
+	align-items: center;
+	display: flex;
+	gap: $gap;
+	justify-content: flex-end;
+	margin-top: $gap;
+
+	// The specificity here is needed to overwrite the margin-top that is inherited on WC block template pages such as Shop.
+	button[type="submit"]:not(.wp-block-search__button).wc-block-components-filter-submit-button {
```
Comment (tjcafferkey) [inline assets/js/blocks/rating-filter/style.scss:36]: I think the following is needed for disabled button styling.

```css
&:disabled {
    opacity: .6;
    cursor: auto;
}
```

File: assets/js/blocks/rating-filter/block.tsx [hunk 2]
```diff
@@ -239,15 +287,40 @@ const RatingFilterBlock = ( {
...
+								'Reset rating filter',
+								'woo-gutenberg-products-block'
+							) }
+						/>
+					) }
+					{ blockAttributes.showFilterButton && (
+						<FilterSubmitButton
+							className="wc-block-rating-filter__button"
+							isLoading={ isLoading }
+							disabled={ isLoading || isDisabled }
```
Comment (tjcafferkey) [inline assets/js/blocks/rating-filter/block.tsx:318]:
```suggestion
							disabled={ isLoading || isDisabled || clicked.length === 0 }
```

**Comment by danieldudzic** [comment]
>     1. When I add the filter to a template Site Editor > All Templates > Product Catalog I don't get the icons in the Editor like I do on a regular page alongside the All Products block. The icons work on the frontend though.
> 
I have already created an issue for this: https://github.com/woocommerce/woocommerce-blocks/issues/7167 - I believe it's unrelated to the filter block :)

**Comment by danieldudzic** [comment]
@tjcafferkey Thank you for the review! I have now addressed your feedback in points 2 & 3.

**Comment by tjcafferkey** [comment]
Screenshot 2022-10-10 at 15 39 33

It looks like when I have selected a rating to filter by, the star icons change color and also the number of items are bolded. This isn't consistent with the other blocks (attribute / stock) as they don't change color or become bold when selected.

**Comment by danieldudzic** [comment]
> Screenshot 2022-10-10 at 15 39 33
> 
> It looks like when I have selected a rating to filter by, the star icons change color and also the number of items are bolded. This isn't consistent with the other blocks (attribute / stock) as they don't change color or become bold when selected.

Right, that styling has been added based on @vivialice's suggestion, here:  pdnLyh-1sR-p2#comment-1044

**Comment by tjcafferkey** [comment]
>Right, that styling has been added based on @vivialice's suggestion, here: pdnLyh-1sR-p2#comment-1044

Thanks for clearing that up. Also, it is probably unrelated to the PR. One other issue I've found is within the Active Filters block. You can see that two separate headings are created when you filter by two different ratings, but they are kept under the same heading for attributes which is how it should be imo.

Screenshot 2022-10-10 at 16 05 00

Again, this is another issue unrelated to this PR. I'm happy for this to be merged but we should probably open a followup issue for the above.

**Review by tjcafferkey — APPROVED** [review]

File: assets/js/blocks/rating-filter/block.tsx [hunk 3]
```diff
@@ -240,14 +283,39 @@ const RatingFilterBlock = ( {
...
 					isDisabled={ isDisabled }
 				/>
 			</div>
+			{
+				<div className="wc-block-rating-filter__actions">
+					{ checked.length > 0 && ! isLoading && (
+						<FilterResetButton
+							onClick={ () => {
+								setChecked( [] );
+								setProductRatings( [] );
```
Comment (roykho) [inline assets/js/blocks/rating-filter/block.tsx:300]: Hi @danieldudzic - I recently ran into an error of undefined function `setProductRatings`. Is this a typo and should actually be `setProductRatingsQuery`?

Comment (tjcafferkey) [reply assets/js/blocks/rating-filter/block.tsx:300]: @roykho can you please provide some replication steps for this error? :)

Comment (tjcafferkey) [reply assets/js/blocks/rating-filter/block.tsx:300]: I've created an issue and opened a PR for this. Good catch @roykho !

https://github.com/woocommerce/woocommerce-blocks/issues/8373

**Pull request merged**